### PR TITLE
initial implementation of code generation

### LINKED ISSUE #3: add support for code generation for multiple schemas at a time
support generating code for "large" collections of schemas (~thousands of avsc files)

### PR DESCRIPTION
**Pull request by radai-rosenblatt**
closes #3
Signed-off-by: radai-rosenblatt <radai.rosenblatt@gmail.com>

### CODE REVIEW

File: avro-codegen/src/main/java/com/linkedin/avro/codegen/CodeGenerator.java [hunk 1]
```diff
@@ -0,0 +1,775 @@
...
+            Map<File, FileParseIssue> failedFiles = new HashMap<>();
+
+            for (File schemaFile : toTry) {
+                String fileContents = readFile(schemaFile);
+                SchemaParseResult result;
+                try {
+
+                    //build up a collection of all known schemas to hand to avro
+                    allKnownSchemas.clear();
+                    allKnownTopLevelSchemas.clear();
```
Comment (radai-rosenblatt) [inline avro-codegen/src/main/java/com/linkedin/avro/codegen/CodeGenerator.java]: not required

File: avro-codegen/src/main/java/com/linkedin/avro/codegen/CodeGenerator.java [hunk 2]
```diff
@@ -0,0 +1,775 @@
...
+            Map<String, SchemaDetails> moreSchemas,
+            Map<String, SchemaDetails> externalSchemas
+    ) {
+
+//        Schema mainSchema = result.getMainSchema();
+        Map<String, Schema> allSchemas = result.getAllSchemas(); //will contain all schemas defined anywhere
+
+        Map<String, Schema> actuallyNewSchemas = new HashMap<>();
+
+//        String mainSchemaFullName = mainSchema.getFullName();
```
Comment (radai-rosenblatt) [inline avro-codegen/src/main/java/com/linkedin/avro/codegen/CodeGenerator.java]: oops

File: avro-codegen/src/main/java/com/linkedin/avro/codegen/CodeGenerator.java [hunk 3]
```diff
@@ -0,0 +1,775 @@
...
+
+        return actuallyNewSchemas;
+    }
+
+    private void buildUpKnownSchemaLists(
+            Map<String, SchemaDetails> successfullyParsed,
+            Map<String, SchemaDetails> moreSchemas,
+            Map<String, SchemaDetails> externalSchemas,
+            List<Schema> allKnownSchemas,
+            List<Schema> allKnownTopLevelSchemas
```
Comment (radai-rosenblatt) [inline avro-codegen/src/main/java/com/linkedin/avro/codegen/CodeGenerator.java]: can be removed

File: avro-codegen/src/main/java/com/linkedin/avro/codegen/CodeGenerator.java [hunk 4]
```diff
@@ -0,0 +1,775 @@
...
+                    File file = new File(input, included);
+                    includeFile(file, input, fileToParent, sharedSchemas, nonSharedSchemas);
+                }
+            } else {
+                LOG.warn("path {} is neither a file nor a directory", input);
+            }
+        }
+    }
+
+    private void includeFile(File file, File root, Map<File, File> fileToParent, Set<File> sharedSchemas, Set<File> nonSharedSchemas) {
```
Comment (jimhe) [inline avro-codegen/src/main/java/com/linkedin/avro/codegen/CodeGenerator.java]: If I'm understanding this correctly, I think `fileToParent` might be more clearly named as `fileToRoot`?

**Pull request merged**